### PR TITLE
Fixed retrying consent screen error on cancelling

### PR DESCRIPTION
**Pull request by danishnavid008**
isseu #3228

### CODE REVIEW

File: extension/js/common/api/email-provider/gmail/google-auth.ts [hunk 1]
```diff
@@ -129,14 +129,9 @@ export class GoogleAuth {
       GoogleAuth.waitForAndProcessOauthWindowResult(oauthWin.id, acctEmail, scopes, authRequest.csrfToken, save),
       GoogleAuth.waitForOauthWindowClosed(oauthWin.id, acctEmail),
     ]);
-    try {
-      chrome.windows.remove(oauthWin.id);
-    } catch (e) {
-      if (String(e).indexOf('No window with id') === -1) {
-        Catch.reportErr(e);
-      }
```
Comment (tomholub) [inline extension/js/common/api/email-provider/gmail/google-auth.ts:137]: Why do you propose removing the try/catch? Did it give you any trouble?

**Review by tomholub — CHANGES_REQUESTED** [review]
I prefer that the bug is fixed as suggested in the issue:

> To fix this, you can change the Explaining FlowCrypt webmail permissions modal from confirm to warning and then the button will just close the modal. Remove the sentence Would you like to proceed to the permissions screen again?.

Your solution is not bad, but to make sure it doesn't break again, you'd have to write a test for it too (which is difficult until you get familiarized with our tests). Instead of doing that, it's easier to fix it as suggested.

File: extension/js/common/api/email-provider/gmail/google-auth.ts (hunk 1 above)
Comment (danishnavid008) [reply extension/js/common/api/email-provider/gmail/google-auth.ts:137]: I figured out the error was with `chrome.windows.remove(oauthWin.id);` method. On cancel the chrome window is already removed by itself and we don't have the windows, so it throws the error. What we need to do was to delete the windows once the result is returned from the promise. That way we don't need the try/catch.

Comment (danishnavid008) [reply extension/js/common/api/email-provider/gmail/google-auth.ts:137]: If you want me to fix it by the suggested way I can do it as well.

Comment (tomholub) [reply extension/js/common/api/email-provider/gmail/google-auth.ts:137]: Ok. That does make sense. We can go with your proposal then, but I'll need to fix a failing test - there's a missing await. Take a look. https://flowcrypt.semaphoreci.com/workflows/44c0d9e4-4bc3-49e0-8ebc-9a8ce0c337d1

**Review by tomholub — APPROVED** [review]

**Comment by tomholub** [comment]
It's missing a semicolon. These type of issues should be auto-fixed on every save when you open the workspace using the workspace file thanks to these settings:

```
    "editor.formatOnSave": true,
    "editor.formatOnPaste": true,
```

Maybe you have some personal settings that over-rule this?

**Comment by danishnavid008** [comment]
> It's missing a semicolon. These type of issues should be auto-fixed on every save when you open the workspace using the workspace file thanks to these settings:
> 
> ```
>     "editor.formatOnSave": true,
>     "editor.formatOnPaste": true,
> ```
> 
> Maybe you have some personal settings that over-rule this?

@tomholub Sorry! I'll fix it and update the PR. I had opened the app using the workspace file and yeah it might be some personal settings that over-rule this.

**Review by tomholub — APPROVED** [review]

**Pull request merged**